File: src/org/uwi/EntropyCanvas.java
package org.uwi;

import java.awt.Color;
import java.awt.Dimension;
import java.awt.Graphics;
import java.awt.Point;

import javax.swing.JPanel;

// SwingJS  moved from Canvas to JPanel

public class EntropyCanvas extends JPanel {
  public double Entropy[];
  public int EntropyCalc;
  
  public EntropyCanvas()   {
    super();
  }

  public void paint(Graphics g)   {
    g.setColor(Color.blue);
    Dimension dim = this.getSize();
    double xScale = dim.width/EntropyCalc;
    double yScale = (dim.height-20)/(Entropy[EntropyCalc-1] + 10);
    Point start = new Point();
    start.x = 0;
    start.y = dim.height;
    for (int i=0;i<50;i++)     {
      g.drawLine(start.x,start.y,(int)Math.floor(i * xScale), dim.height - ((int) Math.floor(Entropy[i] * yScale)));
      start.x = (int)Math.floor(i * xScale);
      start.y = dim.height - ((int) Math.floor(Entropy[i] * yScale));
    }
  }
  
}

File: site/swingjs/j2s/org/uwi/EntropyCanvas.js
Clazz.declarePackage ("org.uwi");
Clazz.load (["javax.swing.JPanel"], "org.uwi.EntropyCanvas", ["java.awt.Color", "$.Point"], function () {
c$ = Clazz.decorateAsClass (function () {
this.Entropy = null;
this.EntropyCalc = 0;
this.entCounter = 0;
Clazz.instantialize (this, arguments);
}, org.uwi, "EntropyCanvas", javax.swing.JPanel);
Clazz.overrideMethod (c$, "paintComponent", 
function (g) {
g.setColor (java.awt.Color.blue);
var dim = this.getSize ();
var xScale = Clazz.doubleToInt (dim.width / this.EntropyCalc);
var yScale = (dim.height - 20) / (org.uwi.EntropyCanvas.eMax + 10);
var start =  new java.awt.Point ();
start.x = 0;
start.y = dim.height;
for (var i = 0; i < this.entCounter; i++) {
var x = Clazz.doubleToInt (Math.floor (i * xScale));
var y = dim.height - (Clazz.doubleToInt (Math.floor (this.Entropy[i] * yScale)));
g.drawLine (start.x, start.y, x, y);
start.x = x;
start.y = y;
}
}, "java.awt.Graphics");
Clazz.defineStatics (c$,
"eMax", 65000);
});
